### PR TITLE
Add a registry for PLL_Translatable_Object

### PR DESCRIPTION
**Pull request by manooweb**
Related #1149, https://github.com/polylang/polylang-wc/issues/502.

Requires #1159 

`get_terms()` function obviously accept an array of taxonomy names.
See https://developer.wordpress.org/reference/functions/get_terms/#parameters

### CODE REVIEW

**Comment by Screenfeed** [comment]
If my branch is merged with [this change](https://github.com/polylang/polylang/pull/1159/commits/192991bcf8f7995e14811239a4ebac349d9d67ed#diff-889b75dbff02fa326b801140fb1c1e280201bb4131e9ee84a4e0d3770a733620R368), we'll need to change calls to `get_objects_with_no_lang()` here (remove the 1st argument).

**Review by Screenfeed — CHANGES_REQUESTED** [review]

File: include/model.php [hunk 1]
```diff
@@ -98,53 +108,32 @@ public function has_languages() {
...
-					set_transient( 'pll_languages_list', array_map( 'get_object_vars', $languages ) );
-				}
-			}
-
-			// Create the languages directly from arrays stored in transients.
-			else {
-				foreach ( $languages as $k => $v ) {
-					$languages[ $k ] = new PLL_Language( $v );
+					// Re-index.
+					$languages = array_values( $languages );
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php]: Done in 8af440e56cb7c0992eae706288f2ee07a5f4cd14

File: include/model.php [hunk 2]
```diff
@@ -654,10 +609,14 @@ public function get_objects_with_no_lang( $limit = -1 ) {
...
-		$limit = (int) apply_filters( 'get_objects_with_no_lang_limit', $limit );
+		$limit   = apply_filters( 'get_objects_with_no_lang_limit', $limit );
+		$limit   = $limit < 1 ? -1 : max( (int) $limit, 1 );
+		$objects = array();
 
-		$posts = $this->get_posts_with_no_lang( $this->get_translated_post_types(), $limit );
-		$terms = $this->get_terms_with_no_lang( $this->get_translated_taxonomies(), $limit );
+		foreach ( $this->translatable_objects->get_all() as $type => $object ) {
+			// The trailing 's' in the array key is for backward compatibility.
+			$objects[ "{$type}s" ] = $object->get_objects_with_no_lang( $object->get_translated_object_types(), $limit );
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php]: Done in 8af440e56cb7c0992eae706288f2ee07a5f4cd14

File: include/model.php [hunk 3]
```diff
@@ -712,39 +657,13 @@ public function get_posts_with_no_lang( $post_types, $limit ) {
...
-
-		$term_ids = wp_cache_get( $cache_key, 'terms' );
-
-		if ( false === $term_ids ) {
-			$term_ids = $wpdb->get_col( $sql ); // PHPCS:ignore WordPress.DB.PreparedSQL.NotPrepared
-			wp_cache_set( $cache_key, $term_ids, 'terms' );
-		}
-
-		return $term_ids;
+		return $object->get_objects_with_no_lang( (array) $taxonomies, $limit );
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php]: Done in 8af440e56cb7c0992eae706288f2ee07a5f4cd14

File: include/model.php [hunk 4]
```diff
@@ -778,6 +697,136 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
...
+		foreach ( $terms as $term ) {
+			$terms_by_type['post'][ $term->slug ] = $term;
+		}
+
+		$translatable_objects = array_diff_key(
+			$this->translatable_objects->get_all(),
+			$terms_by_type
+		);
+
+		foreach ( $translatable_objects as $type => $sub_model ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php]: Done in 8af440e56cb7c0992eae706288f2ee07a5f4cd14

**Review by Chouby — CHANGES_REQUESTED** [review]

File: include/model.php [hunk 5]
```diff
@@ -218,8 +201,9 @@ public function get_language( $value ) {
 
 		if ( false === $return = $this->cache->get( 'language:' . $value ) ) {
 			foreach ( $this->get_languages_list() as $lang ) {
-				$this->cache->set( 'language:' . $lang->term_id, $lang );
-				$this->cache->set( 'language:' . $lang->tl_term_id, $lang );
+				foreach ( $lang->get_tax_props( 'term_id' ) as $term_id ) {
+					$this->cache->set( 'language:' . $term_id, $lang );
+				}
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php:206]: We decided not to change this.

File: include/translatable-objects-registry.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * @package Polylang
+ */
+
+/**
+ * Registry for all translatable objects.
+ *
+ * @since 3.4
+ */
+class PLL_Translatable_Objects_Registry {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-objects-registry.php]: Done in e9d0e130f115c25d35f2fcfe4048eeebc70e8de1

File: include/translatable-objects-registry.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+	 *
+	 * @since 3.4
+	 *
+	 * @return PLL_Translatable_Object[] Array keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-return array<non-empty-string, PLL_Translatable_Object>
+	 */
+	public function get_all() {
+		return $this->objects;
+	}
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-objects-registry.php]: Done in b0241d463b0db189db5663c21a98123f607a4761 and 22364678abfa7cbc38b96e751b22693265b3af5b

File: include/model.php [hunk 8]
```diff
@@ -794,4 +832,31 @@ protected function get_language_terms() {
...
+	 *
+	 * @since 3.4
+	 *
+	 * @param bool $language     True to return taxonomy which manages language.
+	 * @param bool $translations True to return taxonomy which manages translation.
+	 * @return string[] Taxonomy names.
+	 *
+	 * @phpstan-return list<non-empty-string>
+	 */
+	public function get_taxonomy_names( $language = true, $translations = true ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php]: Done in 100651dc3a730d305ef5d76223763be3a6172b2d

File: admin/admin-model.php [hunk 9]
```diff
@@ -377,7 +398,7 @@ public function set_language_in_mass( $type, $ids, $lang ) {
 	public function set_translation_in_mass( $type, $translations ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php]: Done in 5d110a213d3517674efc898ec536beb68622f839

File: admin/admin-model.php [hunk 10]
```diff
@@ -329,13 +338,24 @@ protected function validate_lang( $args, $lang = null ) {
 	public function set_language_in_mass( $type, $ids, $lang ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php]: Done in 5d110a213d3517674efc898ec536beb68622f839

File: admin/admin-model.php [hunk 11]
```diff
@@ -577,4 +600,17 @@ public function update_default_lang( $slug ) {
...
+	 *
+	 * @since 3.4
+	 *
+	 * @return PLL_Translatable_Object[] An array of secondary translatable objects. Array keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-return array<non-empty-string, PLL_Translatable_Object>
+	 */
+	protected function get_secondary_translatable_objects() {
+		return array_diff_key( $this->translatable_objects->get_all(), array( 'post' => null ) );
+	}
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php]: Done in 37d5dc2fcbcf6fc11adefd0e41c76c490471de28

File: include/model.php [hunk 12]
```diff
@@ -712,39 +651,13 @@ public function get_posts_with_no_lang( $post_types, $limit ) {
 	 * @return int[]
 	 */
 	public function get_terms_with_no_lang( $taxonomies, $limit ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php:653]: > Same as above.

See https://github.com/polylang/polylang/pull/1160#discussion_r1046850636

File: include/translatable-objects.php [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+	/**
+	 * Returns all registered translatable objects.
+	 *
+	 * @since 3.4
+	 *
+	 * @return ArrayIterator Iterator on $objects array property. Keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-return ArrayIterator<string, PLL_Translatable_Object>
```
Comment (Screenfeed) [inline include/translatable-objects.php:47]: Replacing `non-empty-string` by `string` here will probably cause other PHPStan issues elsewhere.
Let's keep it like that for now, we'll see what happens in the future big PHPStan sweep.

File: include/translatable-objects.php [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+	 */
+	protected function get_taxonomy_names( $filter = array( 'language', 'translations' ) ) {
+		$taxonomies = array();
+
+		foreach ( $this->objects as $object ) {
+			if ( in_array( 'language', $filter ) ) {
+				$taxonomies[] = $object->get_tax_language();
+			}
+
+			if ( in_array( 'translations', $filter ) && $object instanceof PLL_Translated_Object ) {
```
Comment (Screenfeed) [inline include/translatable-objects.php:101]: Let's be strict:
```suggestion
			if ( in_array( 'language', $filter, true ) ) {
				$taxonomies[] = $object->get_tax_language();
			}

			if ( in_array( 'translations', $filter, true ) && $object instanceof PLL_Translated_Object ) {
```

File: modules/wizard/wizard.php [hunk 15]
```diff
@@ -704,10 +704,10 @@ public function save_step_untranslated_contents() {
 
 		while ( $nolang = $this->model->get_objects_with_no_lang( 1000 ) ) {
 			if ( ! empty( $nolang['posts'] ) ) {
-				$this->model->set_language_in_mass( 'post', $nolang['posts'], $language->slug );
+				$this->model->post->set_language_in_mass( 'post', $nolang['posts'], $language->slug );
 			}
 			if ( ! empty( $nolang['terms'] ) ) {
-				$this->model->set_language_in_mass( 'term', $nolang['terms'], $language->slug );
+				$this->model->term->set_language_in_mass( 'term', $nolang['terms'], $language->slug );
 			}
```
Comment (Screenfeed) [inline modules/wizard/wizard.php:711]: Here and in `settings.php` we'll need to "set language in mass" for all types, not only for `post` and `term`.
I suggest do do that in another PR, maybe in the one dealing with `get_objects_with_no_lang()`.

**Review by Screenfeed — CHANGES_REQUESTED** [review]

File: include/translatable-objects.php [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+	/**
+	 * Returns all translatable objects except post one.
+	 *
+	 * @since 3.4
+	 *
+	 * @return PLL_Translatable_Object[] An array of secondary translatable objects. Array keys are the type of translated content (post, term, etc).
+	 *
+	 * @phpstan-return array<non-empty-string, PLL_Translatable_Object>
+	 */
+	protected function get_secondary_translatable_objects() {
```
[the remark this answers is not in the document]
Comment (Chouby) [reply include/translatable-objects.php:78]: I agree on the goal. Could we simplify and say the first registered is the main one?

File: modules/wizard/wizard.php (hunk 15 above)
Comment (Chouby) [reply modules/wizard/wizard.php:711]: OK to solve this in another PR.

File: include/translatable-objects.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (Screenfeed) [reply include/translatable-objects.php:78]: Why not. Something like that then:
```php
class PLL_Translatable_Objects implements IteratorAggregate {

	/**
	 * Type of the main translatable object.
	 *
	 * @var string
	 *
	 * @phpstan-var non-empty-string|null
	 */
	private $main_type;

	public function register( PLL_Translatable_Object $object ) {
		if ( empty( $this->main_type ) ) {
			$this->main_type = $object->get_type();
		}

		if ( ! isset( $this->objects[ $object->get_type() ] ) ) {
			$this->objects[ $object->get_type() ] = $object;
		}

		return $this->objects[ $object->get_type() ];
	}

	public function get_secondary_translatable_objects() {
		if ( empty( $this->main_type ) ) {
			return array();
		}

		return array_diff_key( $this->objects, array( $this->main_type => null ) );
	}
}
```

File: modules/wizard/wizard.php (hunk 15 above)
Comment (manooweb) [reply modules/wizard/wizard.php:711]: I completed #1163 issue.

File: include/translatable-objects.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-objects.php:78]: Why not? Couldn't we hard code 'post' to be even clearer?

```php
private $main_type = 'post';
```

File: include/translatable-objects.php [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+	 *
+	 * @since 3.4
+	 *
+	 * @param string[] $filter An array on value to filter taxonomy names to return.
+	 * @return string[] Taxonomy names.
+	 *
+	 * @phpstan-param array<'language'|'translations'> $filter
+	 * @phpstan-return list<non-empty-string>
+	 */
+	protected function get_taxonomy_names( $filter = array( 'language', 'translations' ) ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-objects.php:93]: Done in dd704d12f456141d5837ccd063e917701de78933

File: include/translatable-objects.php (hunk 14 above)
Comment (manooweb) [reply include/translatable-objects.php:101]: Done in e5019bc01e0b443a2e939f410bccf6dc44364f72

File: admin/admin-model.php [hunk 18]
```diff
@@ -227,16 +230,22 @@ public function update_language( $args ) {
...
-		// And finally update the language itself
-		$description = maybe_serialize( array( 'locale' => $args['locale'], 'rtl' => (int) $args['rtl'], 'flag_code' => empty( $args['flag'] ) ? '' : $args['flag'] ) );
-		wp_update_term( (int) $lang->term_id, 'language', array( 'slug' => $slug, 'name' => $args['name'], 'description' => $description, 'term_group' => (int) $args['term_group'] ) );
-		if ( empty( $lang->tl_term_id ) ) {
-			// Attempt to repair the term_language if it has been deleted by a database cleaning tool.
-			wp_insert_term( $args['name'], 'term_language', array( 'slug' => 'pll_' . $slug ) );
-		} else {
-			wp_update_term( (int) $lang->tl_term_id, 'term_language', array( 'slug' => 'pll_' . $slug, 'name' => $args['name'] ) );
+		// And finally update the language itself.
+		foreach ( $this->get_secondary_translatable_objects() as $object ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php:234]: Done in d0a7b9f09514a4d99a9e2806047b8ce5883cc5be

File: include/model.php [hunk 19]
```diff
@@ -778,6 +691,131 @@ public function filter_language_terms_orderby( $orderby, $args, $taxonomies ) {
...
+			return array();
+		}
+
+		$terms_by_type = array();
+
+		foreach ( $terms as $term ) {
+			$terms_by_type['post'][ $term->slug ] = $term;
+		}
+
+		foreach ( $this->get_secondary_translatable_objects() as $type => $sub_model ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/model.php:796]: Done in d0a7b9f09514a4d99a9e2806047b8ce5883cc5be

File: admin/admin-model.php [hunk 20]
```diff
@@ -42,9 +42,10 @@ public function add_language( $args ) {
 		}
 		wp_update_term( (int) $r['term_id'], 'language', array( 'term_group' => (int) $args['term_group'] ) ); // can't set the term group directly in wp_insert_term
 
-		// The term_language taxonomy
-		// Don't want shared terms so use a different slug
-		wp_insert_term( $args['name'], 'term_language', array( 'slug' => 'pll_' . $args['slug'] ) );
+		// The other language taxonomies.
+		foreach ( $this->get_secondary_translatable_objects() as $object ) {
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php:46]: Done in d0a7b9f09514a4d99a9e2806047b8ce5883cc5be

File: include/translatable-object.php [hunk 21]
```diff
@@ -452,4 +452,145 @@ protected function get_objects_with_no_lang_sql( $language_ids, $limit ) {
...
+			return;
+		}
+
+		$lang = $this->get_language( $lang );
+
+		if ( empty( $lang ) ) {
+			return;
+		}
+
+		$tax_language = $this->translatable_objects->get( $type )->get_tax_language();
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-object.php:481]: Done in 36617cb9df1149a66f89c5bd9fa26a7b1a29e318

**Review by Screenfeed — APPROVED** [review]

File: include/translatable-objects.php (hunk 16 above)
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translatable-objects.php]: Done in 05b0d0031f8e1916cab8672823f8c086c51203b4

File: include/translatable-objects.php [hunk 22]
```diff
@@ -12,6 +12,15 @@
  */
 class PLL_Translatable_Objects implements IteratorAggregate {
 
+	/**
+	 * Type of the main translatable object.
+	 *
+	 * @var string
+	 *
+	 * @phpstan-var non-empty-string
```
Comment (Screenfeed) [inline include/translatable-objects.php:20]: Since it is not defined in the constructor:
```suggestion
	 * @phpstan-var non-empty-string|null
```

**Review by Screenfeed — CHANGES_REQUESTED** [review]

File: admin/admin-model.php [hunk 23]
```diff
@@ -289,7 +298,7 @@ protected function validate_lang( $args, $lang = null ) {
 
 		// Validate slug is unique
 		foreach ( $this->get_languages_list() as $language ) {
-			if ( $language->slug === $args['slug'] && ( null === $lang || $lang->term_id !== $language->term_id ) ) {
+			if ( $language->slug === $args['slug'] && ( null === $lang || $lang->get_tax_prop( 'language', 'term_id' ) !== $language->get_tax_prop( 'language', 'term_id' ) ) ) {
```
Comment (Chouby) [inline admin/admin-model.php:301]: No need to change this as we kept the `term_id` property. For me the old way is more readable.

File: admin/admin-model.php [hunk 24]
```diff
@@ -456,11 +338,13 @@ public function set_translation_in_mass( $type, $translations ) {
...
-		$dr       = array();
-		$dt       = array();
-		$ut       = array();
+		$taxonomies = $this->translatable_objects->get_taxonomy_names( array( 'translations' ) );
+		$term_ids   = array();
+		$dr         = array();
+		$dt         = array();
+		$ut         = array();
+
+		$terms = get_terms( array( 'taxonomy' => $taxonomies ) );
```
Comment (Chouby) [inline admin/admin-model.php:347]: To slightly improve the readability:
```suggestion
		$term_ids = array();
		$dr       = array();
		$dt       = array();
		$ut       = array();

		$taxonomies = $this->translatable_objects->get_taxonomy_names( array( 'translations' ) );
		$terms      = get_terms( array( 'taxonomy' => $taxonomies ) );

```

File: include/model.php [hunk 25]
```diff
@@ -98,53 +108,26 @@ public function has_languages() {
...
+			} else {
+				$languages = get_transient( 'pll_languages_list' );
+
+				if ( ! is_array( $languages ) ) {
+					// Create the languages from taxonomies.
+					$languages = $this->get_languages_from_taxonomies();
+				} else {
+					// Create the languages directly from arrays stored in the transient.
+					foreach ( $languages as $k => $v ) {
+						$languages[ $k ] = PLL_Language_Factory::create( $v );
```
Comment (Chouby) [inline include/model.php:126]:
```suggestion
						$languages[ $k ] = PLL_Language_Factory::get( $v );
```

**Review by Chouby — CHANGES_REQUESTED** [review]

File: admin/admin-model.php (hunk 24 above)
Comment (manooweb) [reply admin/admin-model.php:347]: Done in b5d73a8151cf728cf599d09202e2b233bf0af954

File: include/model.php (hunk 25 above)
Comment (manooweb) [reply include/model.php:126]: Done in ebf0060f535ee436b809d597c99f6b8fc9dbdfa1

File: include/translatable-objects.php [hunk 26]
```diff
@@ -0,0 +1,121 @@
...
+class PLL_Translatable_Objects implements IteratorAggregate {
+
+	/**
+	 * Type of the main translatable object.
+	 *
+	 * @var string
+	 *
+	 * @phpstan-var non-empty-string|null
+	 */
+	private $main_type;
```
Comment (Chouby) [inline include/translatable-objects.php:22]: I suggest not changing the type when we can.
```suggestion
	 */
	private $main_type = '';
```

File: include/translatable-object.php [hunk 27]
```diff
@@ -452,4 +452,46 @@ protected function get_objects_with_no_lang_sql( $language_ids, $limit ) {
...
+		global $wpdb;
+
+		$lang = $this->get_language( $lang );
+
+		if ( empty( $lang ) ) {
+			return;
+		}
+
+		$tax_language = $this->get_tax_language();
+		$tt_id        = $lang->get_tax_prop( $tax_language, 'term_taxonomy_id' );
```
Comment (Chouby) [inline include/translatable-object.php:476]: Not neccessary to use the getter in the class itself:
```suggestion
		$tt_id = $lang->get_tax_prop( $this->tax_language, 'term_taxonomy_id' );
```

File: include/translatable-object.php [hunk 28]
```diff
@@ -452,4 +452,46 @@ protected function get_objects_with_no_lang_sql( $language_ids, $limit ) {
...
+	 * @return void
+	 */
+	public function set_language_in_mass( $ids, $lang ) {
+		global $wpdb;
+
+		$lang = $this->get_language( $lang );
+
+		if ( empty( $lang ) ) {
+			return;
+		}
```
Comment (Chouby) [inline include/translatable-object.php:473]: If it doesn't require too many changes, I suggest to accept only `PLL_Language` and not `string`.

File: include/translated-object.php [hunk 29]
```diff
@@ -456,4 +456,82 @@ protected function validate_translations( $translations, $id = 0, $context = 'sa
 		/** @phpstan-var array<non-empty-string, positive-int> */
 		return array_merge( array( $lang->slug => $id ), $translations );
 	}
+	/**
+	 * Creates translations groups in mass.
+	 *
+	 * @since 1.6.3
+	 * @since 3.4 Moved from PLL_Admin_Model class.
+	 *
+	 * @param array  $translations Array of translations arrays.
```
Comment (Chouby) [inline include/translated-object.php:465]: Can we better precise the type? Also remove extra spaces.

File: include/translated-term.php [hunk 30]
```diff
@@ -299,4 +299,28 @@ protected function should_update_translation_group( $id, $translations ) {
...
 	}
+
+	/**
+	 * Assigns a language to terms in mass.
+	 *
+	 * @since 1.2
+	 * @since 3.4 Moved from PLL_Admin_Model class.
+	 *
+	 * @param int[]               $ids  Array of post ids or term ids.
+	 * @param PLL_Language|string $lang Language to assign to the posts or terms.
```
Comment (Chouby) [inline include/translated-term.php:310]: Same as above.

**Review by Chouby — CHANGES_REQUESTED** [review]

File: admin/admin-model.php [hunk 31]
```diff
@@ -456,11 +338,13 @@ public function set_translation_in_mass( $type, $translations ) {
...
 
-		$terms    = get_terms( array( 'taxonomy' => array( 'post_translations', 'term_translations' ) ) );
-		$term_ids = array();
-		$dr       = array();
-		$dt       = array();
-		$ut       = array();
+		$term_ids   = array();
+		$dr         = array();
+		$dt         = array();
+		$ut         = array();
```
[the remark this answers is not in the document]
Comment (manooweb) [reply admin/admin-model.php:344]: Done in b4f11995c057143cdab78505fec86a33e853aaf4

File: include/translated-object.php [hunk 32]
```diff
@@ -456,4 +456,82 @@ protected function validate_translations( $translations, $id = 0, $context = 'sa
...
+	 * @return void
+	 */
+	public function set_translation_in_mass( $translations ) {
+		global $wpdb;
+
+		$type        = $this->get_type();
+		$taxonomy    = "{$type}_translations";
```
[the remark this answers is not in the document]
Comment (manooweb) [reply include/translated-object.php:472]: Done in 4cf7fd74c4f07d5a26fea6c39ea9cd81b92cb883

File: include/translatable-objects.php (hunk 26 above)
Comment (manooweb) [reply include/translatable-objects.php:22]: So $main_type can't be non-empty-string 😁

File: include/translatable-object.php (hunk 27 above)
Comment (manooweb) [reply include/translatable-object.php:476]: Done in d5f6ee7d27ba75515801f5154db53476927b3b96

File: include/translatable-object.php (hunk 28 above)
Comment (manooweb) [reply include/translatable-object.php:473]: 🤔 we can in the wizard calls because we have PLL_Language object.
But what about in `PLL_Settings` calls which uses language slug (default_lang option)?
See https://github.com/polylang/polylang/blob/3.3.x/settings/settings.php#L244-L249

File: include/translated-term.php (hunk 30 above)
Comment (manooweb) [reply include/translated-term.php:310]: Same question as above 🙂

File: include/translated-object.php (hunk 29 above)
Comment (manooweb) [reply include/translated-object.php:465]: Done in a56a670593dda2e4ac16f94fcdf079b4086ce5bc

File: include/translated-term.php (hunk 30 above)
Comment (manooweb) [reply include/translated-term.php:310]: Done in 8b867bc1174b361bcae57de13b21591d2ad67e0c

File: include/translatable-object.php (hunk 28 above)
Comment (manooweb) [reply include/translatable-object.php:473]: Done in 8b867bc1174b361bcae57de13b21591d2ad67e0c
